editor/questionbankcontroller.js

```diff
@@ -82,12 +84,20 @@ angular.module('createquestionapp', [])
     };
 
     ecEditor.addEventListener('editor:form:change', function(event, data) {
-      $scope.filterObj.concepts = [];
-      if(data.key == "concepts")
-        _.forEach(data.value, function(dataid) {
+      if (data.templateId == "filterMetaDataTemplate") {
+        if (data.key == "concepts") {
+          $scope.filterObj.concepts = [];
+          _.forEach(data.value, function(dataid) {
             $scope.filterObj.concepts.push(dataid.identifier);
-        });
-      $scope.searchQuestions($scope.filterObj);
+          });
+        } else if (data.key == "topic") {
+          $scope.filterObj.topic = [];
+          _.forEach(data.value, function(dataid) {
+            $scope.filterObj.topic.push(dataid);
+          });
+        }
+        $scope.searchQuestions($scope.filterObj);
```

Comment:
Why are you passing scope level object(filterObj) as a function parameter? It can be accessed directly in any function within the same scope.